lib/autocomplete.js

```diff
@@ -0,0 +1,90 @@
+const _ = require("lodash");
+const aws = require("aws-sdk");
+const parsers = require("./parsers");
+const core = require("./core");
+const consts = require("./consts.json");
+
+/**
+ * Maps parameters passed to Autocomplete function to an object with parsed values.
+ *
+ * @param params: [ { value: any, name: string, type: string } ]
+ * Array of values passed to autocomplete function as action parameters.
+ *
+ * @returns {*} - object with params names as keys and parsed values
+ */
+function parseActionParameters(params) {
+  return params.reduce((acc, {
+    value, name, type, valueType,
+  }) => ({
+    ...acc,
+    [name]: parsers.resolveParser(type || valueType)(value),
+  }), {});
+}
+
+function handleInput(awsService, actionParams, pluginSettings) {
+  const [params, settings] = [actionParams, pluginSettings].map(parseActionParameters);
+  const client = core.getServiceInstance(awsService, params, settings);
+  return { params, settings, client };
+}
+
+function simpleItemsFilter(autocompleteItems, query) {
+  let result = autocompleteItems;
+  if (query) {
+    const qWords = query.split(/[. ]/g).map((word) => word.toLowerCase());
+    result = result.filter((item) => qWords
+      .every((word) => item.value.toLowerCase().includes(word)))
+      .sort((word1, word2) => word1.value.toLowerCase().indexOf(qWords[0])
+          - word2.value.toLowerCase().indexOf(qWords[0]));
+  } else {
+    result = result.sort((a, b) => (a.value >= b.value ? 1 : -1));
+  }
+
+  return result.splice(0, consts.MAX_AUTOCOMPLETE_RESULTS);
+}
+
+function simpleList(awsService, listFuncName, pathToArray, pathToValue) {
```

Comment:
The function implies that it's either a constructor function or a function returning an object, but in fact this function returns another function.

Reply:
Not gonna lie - this part of code is straight up broken and should never get to the review like this. I'm very sorry for that, please disregard this function completely.